### ISSUE TITLE
Retire `KeybindingContext`s

### ISSUE THREAD

**Issue by tsmaeder** [eclipse-theia/theia#12657]
### Feature Description:
While the `context` field in `Keybinding` is deprecated, we're still using instances of `KeybindingContext` internally. We should remove those contexts and rely on `when` expressions in the keybindings. Once that is done, we should remove the `KeybindingContext` API. The `context` field has been deprecated for three years.
The reason for this request is that the `KeybindingContext` interface does not offer change notification. Therefore, we cannot update the UI when the active keybindings change. The alternative would be to add change notification to `KeybindingContext`, but I don't think we should evolve deprecated UI. See also https://github.com/eclipse-theia/theia/issues/12656

**Comment by vince-fugnitto** [comment]
I believe it is a duplicate of https://github.com/eclipse-theia/theia/issues/7849, we can close either or.
